### PR TITLE
cpp: test-hello: add hello world test

### PR DESCRIPTION
**Pull request by maska989**
JIRA: CI-355

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [X] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing linter checks and tests passed.
- [X] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by nalajcie** [comment]
can't we just use `_user/hellocpp/`?

**Comment by damianloew** [comment]
> can't we just use `_user/hellocpp/`?

Yeah, I'd also prefer this approach.

File: cpp/test.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+def harness(p):
+    assert "Hello world++!" in p.readline()
```
Comment (mateusz-bloch) [inline cpp/test.py:2]:
```suggestion
    assert "Hello World++!" in p.readline()
```

alternatively (personally prefer the fail message in this case, if occure):

```suggestion
    p.expect_exact('Hello World++!')
```

File: Makefile.host [hunk 2]
```diff
@@ -1,3 +1,4 @@
 DEFAULT_COMPONENTS = $(filter test_meterfs_%,$(ALL_COMPONENTS))
 DEFAULT_COMPONENTS += $(SAMPLE_TESTS)
 DEFAULT_COMPONENTS += $(LIBC_UNIT_TESTS)
+DEFAULT_COMPONENTS += $(CPP_TESTS)
```
Comment (mateusz-bloch) [inline Makefile.host:4]: The binary is not currently building for host-generic-pc
```suggestion
```

File: cpp/test.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+test:
+    tests:
+        - name: main
+          harness: test.py
+          execute: ../usr/bin/hellocpp
+          target:
```
Comment (mateusz-bloch) [inline cpp/test.yaml:6]:
```suggestion
          targets:
```

File: cpp/test.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+test:
+    tests:
+        - name: main
+          harness: test.py
+          execute: ../usr/bin/hellocpp
```
[the remark this answers is not in the document]
Comment (damianloew) [reply cpp/test.yaml:5]: After some thinking, I'd consider changing the way how this test name is parsed - we could for example check whether there is `\` at the beginning of string and then assume it's full path and don't add `/bin` prefix. This way we could place there just `/usr/bin`, which is much more readable.

File: cpp/test.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+test:
+    tests:
+        - name: main
+          harness: test.py
```
Comment (mateusz-bloch) [inline cpp/test.yaml:4]: Maybe, I would  consider changing the name of the test from `main` to something else, the same with `test.py`

File: cpp/test.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+test:
+
+  tests:
+    - name: test-echo-cpp
+      harness: test-echo.py
```
Comment (damianloew) [inline cpp/test.yaml:5]: Why test-echo? You are not sending anything, I'd leave the same structure as in psh tests:
```suggestion
    - name: hello-cpp
      harness: test-hello-cpp.py
```

**Review by damianloew — COMMENTED** [review]
Moreover, this `test-hello:` is unnecessary in the commit message, the following message would be fine: `cpp: add hello world test`

File: cpp/test.yaml (hunk 6 above)
Comment (maska989) [reply cpp/test.yaml]: My mistake, I made changes other than I intended.

**Review by damianloew — APPROVED** [review]

**Pull request merged**